### PR TITLE
feat: allow multiple approves in case of wrong wrap token

### PR DESCRIPTION
**Pull request by zhigang1992**

### CODE REVIEW

File: clarity/contracts/helpers/amm-swap-pool-helper.clar [hunk 1]
```diff
@@ -100,7 +100,6 @@
             (request-details (try! (get-request-or-fail request-id)))
             (updated-request-details (merge request-details { token-y: wrapped-token-y, status: APPROVED, status-memo: memo })))
         (try! (check-is-approved))
-        (asserts! (is-eq (get status request-details) PENDING) err-request-already-processed)
         (map-set requests request-id updated-request-details)
```
Comment (zhigang1992) [inline clarity/contracts/helpers/amm-swap-pool-helper.clar:104]: We removed this check so we can potentially approve multiple times with different wrap token

File: clarity/contracts/helpers/amm-swap-pool-helper.clar [hunk 2]
```diff
@@ -110,7 +109,7 @@
             (request-details (try! (get-request-or-fail request-id)))
             (updated-request-details (merge request-details { status: REJECTED, status-memo: memo })))
         (try! (check-is-approved))
-        (asserts! (is-eq (get status request-details) PENDING) err-request-already-processed)
```
Comment (zhigang1992) [inline clarity/contracts/helpers/amm-swap-pool-helper.clar:113]: We can still reject a token before it's finalized

File: clarity/contracts/helpers/amm-swap-pool-helper.clar [hunk 3]
```diff
@@ -110,7 +109,7 @@
             (request-details (try! (get-request-or-fail request-id)))
             (updated-request-details (merge request-details { status: REJECTED, status-memo: memo })))
         (try! (check-is-approved))
```
Comment (zhigang1992) [inline clarity/contracts/helpers/amm-swap-pool-helper.clar]: @fiftyeightandeight should we allow user to reject / revoke this themselves?

File: clarity/contracts/helpers/amm-swap-pool-helper.clar (hunk 1 above)
Comment (fiftyeightandeight) [reply clarity/contracts/helpers/amm-swap-pool-helper.clar:104]: Different wrap token should require a new request, no?

File: clarity/contracts/helpers/amm-swap-pool-helper.clar (hunk 2 above)
Comment (fiftyeightandeight) [reply clarity/contracts/helpers/amm-swap-pool-helper.clar:113]: Let me add to check FINALIZED.

File: clarity/contracts/helpers/amm-swap-pool-helper.clar (hunk 3 above)
Comment (fiftyeightandeight) [reply clarity/contracts/helpers/amm-swap-pool-helper.clar]: Yes, that's a good idea. Let's allow users to revoke their own request.

File: clarity/contracts/helpers/amm-swap-pool-helper.clar (hunk 1 above)
Comment (fiftyeightandeight) [reply clarity/contracts/helpers/amm-swap-pool-helper.clar:104]: I added `(asserts! (and (not (is-eq (get status request-details) FINALIZED)) (not (is-eq (get status request-details) REJECTED))) err-request-already-processed)`

File: clarity/contracts/helpers/amm-swap-pool-helper.clar (hunk 3 above)
Comment (fiftyeightandeight) [reply clarity/contracts/helpers/amm-swap-pool-helper.clar]: https://github.com/alexgo-io/alex-v1/pull/770/commits/9ffb942bfaa383bf85e61fa691fccb5fbeb8d192

File: clarity/contracts/helpers/amm-swap-pool-helper.clar (hunk 2 above)
Comment (fiftyeightandeight) [reply clarity/contracts/helpers/amm-swap-pool-helper.clar:113]: https://github.com/alexgo-io/alex-v1/pull/770/commits/9ffb942bfaa383bf85e61fa691fccb5fbeb8d192

File: clarity/contracts/helpers/amm-swap-pool-helper.clar (hunk 1 above)
Comment (fiftyeightandeight) [reply clarity/contracts/helpers/amm-swap-pool-helper.clar:104]: https://github.com/alexgo-io/alex-v1/pull/770/commits/9ffb942bfaa383bf85e61fa691fccb5fbeb8d192

**Pull request merged**